strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java

```diff
@@ -222,12 +224,19 @@ private void validateUploadedChecksumAgainstCache(byte[] checksum,
 
         if (!matchesChecksum(checksum, artifactBasePath, checksumExtension))
         {
+            String checksumStr = new String(checksum, StandardCharsets.UTF_8);
+
             logger.error("The checksum for {} [{}] is invalid!",
                          artifactPath,
-                         new String(checksum, StandardCharsets.UTF_8));
-        }
+                         checksumStr);
+
+            if (repository.isStrictChecksum())
+            {
+                throw new ArtifactStorageException(String.format("Checksum broken [%s] for artifact [%s]", checksumStr, artifactPath));
```

Comment:
Could you please change this to:
```
                throw new ArtifactStorageException(String.format("Invalid checksum [%s] for artifact [%s]", checksumStr, artifactPath));
```
Thanks! :)

Reply:
Done!